adapter/pkg/messaging/event_types.go

```diff
@@ -104,22 +104,26 @@ type Event struct {
 
 // APIEvent for struct API events
 type APIEvent struct {
-	APIID         int      `json:"apiId"`
-	UUID          string   `json:"uuid"`
-	GatewayLabels []string `json:"gatewayLabels"`
-	APIVersion    string   `json:"apiVersion"`
-	APIContext    string   `json:"apiContext"`
-	APIName       string   `json:"apiName"`
-	APIProvider   string   `json:"apiProvider"`
-	APIStatus     string   `json:"apiStatus"`
-	APIType       string   `json:"apiType"`
+	APIID               int                        `json:"apiId"`
+	UUID                string                     `json:"uuid"`
+	GatewayLabels       []string                   `json:"gatewayLabels"`
+	EnvToDataPlaneIDMap GatewayEnvToDataPlaneIDMap `json:"envToDataPlaneIdMap"`
+	APIVersion          string                     `json:"apiVersion"`
+	APIContext          string                     `json:"apiContext"`
+	APIName             string                     `json:"apiName"`
+	APIProvider         string                     `json:"apiProvider"`
+	APIStatus           string                     `json:"apiStatus"`
+	APIType             string                     `json:"apiType"`
 	Event
 	// TODO: (VirajSalaka) Remove this when the event is fixed from APIM side
 	Version string `json:"version"`
 	Context string `json:"context"`
 	Name    string `json:"name"`
 }
 
+// GatewayEnvToDataPlaneIDMap is used to represent gateway environment to data-plane ID mapping in APIEvent
+type GatewayEnvToDataPlaneIDMap map[string]string
```

Comment:
don't need this, we may use `map[string]string` directly inside APIEvent

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/3212/commits/7d325deba6d2def0dcc0baadc01b2db643061ee2